Cesium.CodeGen/Contexts/VariableInfo.cs

```diff
@@ -6,4 +6,6 @@ namespace Cesium.CodeGen.Contexts;
 
 internal record VariableInfo(string Identifier, StorageClass StorageClass, IType Type, IExpression? Constant)
 {
+    static int CurrentIndex = 0;
```

Comment:
_(Note)_

I'd like to have some sort of indexing that's tied to the translation unit, or something.

Such a counter is too error-prone: it will easily overflow on 2 billion variables constructed in a single compilation session. Moreover, this construct makes it impossible to copy a `VariableInfo`.

I am not against having this for now, but we'll need to get rid of it eventually.

Reply:
#634.